### ISSUE TITLE
greaterThanOrEqual and lessThanOrEqual give incorrect values and have no tests

### ISSUE THREAD

**Issue by thedavidmeister** [mathiasverraes/money#99]
Just stumbled upon this when writing tests for my own project.

greaterThanOrEqual() and lessThanOrEqual() give the inverse results to what you would expect.

Given compare():

````
    public function compare(Money $other)
    {
        $this->assertSameCurrency($other);
        if ($this->amount < $other->amount) {
            return -1;
        } elseif ($this->amount == $other->amount) {
            return 0;
        } else {
            return 1;
        }
    }
````

Then we'd expect gtoe:

````
    public function greaterThanOrEqual(Money $other)
    {
        return 0 <= $this->compare($other);
    }
````

and ltoe:

````
    public function lessThanOrEqual(Money $other)
    {
        return 0 >= $this->compare($other);
    }
````

Instead, actual gtoe:

````
    /**
     * @param \Money\Money $other
     * @return bool
     */
    public function greaterThanOrEqual(Money $other)
    {
        return 0 >= $this->compare($other);
    }
````

actual ltoe:

````
    /**
     * @param \Money\Money $other
     * @return bool
     */
    public function lessThanOrEqual(Money $other)
    {
        return 0 <= $this->compare($other);
    }
````

Additionally, there are no tests for this behaviour, which is likely the only reason this was allowed to happen at all:

````
    public function testComparison()
    {
        $euro1 = new Money(1, new Currency('EUR'));
        $euro2 = new Money(2, new Currency('EUR'));
        $usd = new Money(1, new Currency('USD'));

        $this->assertTrue($euro2->greaterThan($euro1));
        $this->assertFalse($euro1->greaterThan($euro2));
        $this->assertTrue($euro1->lessThan($euro2));
        $this->assertFalse($euro2->lessThan($euro1));

        $this->assertEquals(-1, $euro1->compare($euro2));
        $this->assertEquals(1, $euro2->compare($euro1));
        $this->assertEquals(0, $euro1->compare($euro1));
    }
````

**Comment by thedavidmeister** [comment]
https://github.com/mathiasverraes/money/pull/100

**Comment by sagikazarmark** [comment]
Fixed in #100 and #130
